src/datachain/data_storage/warehouse.py

```diff
@@ -343,6 +343,8 @@ def dataset_rows_select_from_ids(
         if (id_col := get_query_id_column(query)) is None:
             raise RuntimeError("sys__id column not found in query")
 
+        query = query._clone().offset(None).limit(None).order_by(None)
```

Comment:
The main change is here. Do not apply any limit, offset or order by if selecting by ids.